src/contoroller/user/user.js

```diff
@@ -8,70 +8,81 @@ class UserController {
-      where: { email: req.body.email },
+      where: { email: email },
     });
     if (user) return res.status(400).send("User already registered.");
-    //Destructure req.body
-    let { username, password, email } = req.body;
-
-    //hassh the password
+    //hashing the password
     const salt = await bcrypt.genSalt(10);
     password = await bcrypt.hash(password, salt);
-    //Save user to the DB
+    //Save new user to the DB
     user = await this.User.create({
       username: username,
       password: password,
       email: email,
     });
+    //Generatting token
     const token = user.generateAuthToken();
     user.token = token;
-
+    //Response to client
     res
       .header("x-auth-token", token)
-      .send(_.pick(user, ["id", "username", "email", "deletedAt", "token"]));
+      .send(_.pick(user, ["id", "username", "email", "token"]));
   }
 
   //
   async update(req, res) {
-    //Look up for the user by given id
-    const user = await this.User.findOne({ where: { id: req.params.id } });
-    if (!user) return res.status(404).send("The user was not found");
-    //Validate received data to update a user
-    const { error } = this.validateUser(req.body);
-    if (error) return res.status(400).send(error.message);
-    //Update user with sent data
-    let { username, password, email } = req.body;
-    const salt = await bcrypt.genSalt(10);
-    password = await bcrypt.hash(password, salt);
-    user.username = username;
-    user.password = password;
-    user.email = email;
-    await user.save();
-    res.send(_.pick(user, ["id", "username", "email", "deletedAt"]));
+    //Find the authorized user
+    const user = await this.User.findOne({ where: { id: req.user.id } });
+    if (!user) {
+      return res.send("user not founded");
+    }
+    return res.send("hey");
+    // //Look up for the user by given id
+    // const user = await this.User.findOne({ where: { id: req.params.id } });
+    // if (!user) return res.status(404).send("The user was not found");
+    // //Validate received data to update a user
+    // const { error } = this.validateUser(req.body);
+    // if (error) return res.status(400).send(error.message);
+    // //Update user with sent data
+    // let { username, password, email } = req.body;
+    // const salt = await bcrypt.genSalt(10);
+    // password = await bcrypt.hash(password, salt);
+    // user.username = username;
+    // user.password = password;
+    // user.email = email;
+    // await user.save();
+    // res.send(_.pick(user, ["id", "username", "email", "deletedAt"]));
   } //!!!
 
-  //
+  //ok
   async delete(req, res) {
-    //Look up for the user by given id
-    const user = await this.User.findOne({ where: { id: req.params.id } });
-    if (!user) return res.status(404).send("The user was not found");
+    //Find the authorized user
+    const user = await this.User.findOne({ where: { id: req.user.id } });
+    if (!user) {
+      return res.send("user not founded");
```

Comment:
You can just return a successful response here, res.status(204). In this way you do not give information about the existence of an entity in your database to the malicious users and since the entity does not exist in the database it can be interpreted as it is deleted.

Reply:
done